wordpress_connector/models/contracts.py

```diff
@@ -458,7 +518,42 @@ def _parse_csp_info(self, data_string):
         # Return a dictionary with the extracted information
         return {
             "country": html2plaintext(country),
+            "continent": CONTINENT_MAPPING.get(html2plaintext(continent), "Africa"),
             "engagement_type": html2plaintext(engagement_type),
             "email": html2plaintext(email),
             "sponsorship_length": html2plaintext(sponsorship_length),
         }
+
+    def _get_neediest_csp_country_code(self, continent=None):
+        """
+        Sort CSP products to find the country with the most needs.
+
+        The neediest country is determined by two factors, in order of priority:
+        1. The lowest percentage of used slots (prioritizing need).
+        2. The highest number of absolute available slots (prioritizing capacity).
+
+        Args:
+            continent (str, optional): The continent ('Africa', 'Asia', 'LATAM')
+                                       to search within. If None, searches globally.
+        Returns:
+            str: The two-letter country code (e.g., 'UG') of the neediest country
+        """
+        if continent:
+            country_codes = CONTINENT_COUNTRY_MAP[continent]
+        else:
+            country_codes = [
+                code
+                for country_list in CONTINENT_COUNTRY_MAP.values()
+                for code in country_list
+            ]
+
+        csp_products = self.env["product.product"].search(
+            [("default_code", "in", [f"csp_{code}" for code in country_codes])]
+        )
+
+        # Sort csp_products by slot_used asc and survival_slot_number desc
+        csp_sorted_by_needs = sorted(
+            csp_products, key=lambda c: (c.slot_used, -c.survival_slot_number)
+        )
+
+        return csp_sorted_by_needs[0].default_code.split("_")[-1]
```

Comment:
Since `slot_used` and `survival_slot_number` are stored, doing all in the search method will be more efficient.
```suggestion
        csp_sorted_by_needs = self.env["product.product"].search(
            [("default_code", "in", [f"csp_{code}" for code in country_codes])],
            order="slot_used ASC, survival_slot_number DESC",
            limit=1
        )
        return csp_sorted_by_needs.default_code.split("_")[-1]
```

Reply:
Thank you, that's way better